[SHni99] ip

SHni99 opened the pull request:
# DUKE  *by Shni99*

> Eat, Sleep, Code, Repeat - Shni99 [Source](https://youtu.be/H4dGpz6cnHo)

Duke is an easy management tool to keep track of your **daily** tasks on hand
- Text-based
- Easy to understand
- ~~boomer~~ Gen Z interface

All you need to do is, 
1. Download it from here
2. Add to your tasks
3. let it manage the tasks for you 🤣 

### Features
- [x] Managing tasks
- [x] Managing deadline
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` [tasklist]
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

SHni99 committed:
Add level one increment

SHni99 committed:
Update adding of items to list

SHni99 committed:
Update list ability to mark and unmark items from the list

SHni99 committed:
Add different kinds of task to the list

SHni99 committed:
Implement automated testing

SHni99 committed:
Handle errors using exceptions

SHni99 committed:
Add delete function to the Duke

SHni99 committed:
Create enum list to store instructions

SHni99 committed:
Add documentation for each classes and methods

SHni99 committed:
Merge branch 'branch-Level-7' and implement Level-7

SHni99 committed:
Merge branch 'branch-Level-7' and implement Level-7

SHni99 committed:
Implement Level-8

SHni99 committed:
Implement Level-8

SHni99 committed:
Merge branch 'branch-Level-8'

SHni99 committed:
Merge branch 'branch-Level-8'

SHni99 committed:
Implement storage class

SHni99 committed:
Add task classes

SHni99 committed:
Implement ui class

SHni99 committed:
Implement tasklist and group tasks in tasks folder

SHni99 committed:
Add parser class

SHni99 committed:
Implement parser class and command classes

SHni99 committed:
Change bye command text

SHni99 committed:
Implement A-Packages

SHni99 committed:
Merge branch 'add-gradle-support'

SHni99 committed:
Implement A-Gradle

SHni99 committed:
Implement A-JUnit and add package to main duke class

SHni99 committed:
Implement A-Jar

SHni99 committed:
Change classname in gradle build

SHni99 committed:
Implement A-JavaDoc

SHni99 committed:
Implement Level-9

BagDownB3nny commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,27 @@
+     * Exits duke if it detects bye command
+     */
+    public abstract boolean isExit();
+
```
Perhaps you could have isExit() normally return false, but only override it in ByeCommand to return true?

BagDownB3nny commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,46 @@
+public class DeleteCommand extends Command {
+    private int index;
+    private boolean exit;
+
```
What is the purpose of `boolean exit`?

BagDownB3nny commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,44 @@
+public class Task {
+    private final String name;
+    private boolean checkMark;
+
```
Perhaps the `checkMark` boolean could be renamed to better suit the coding standards.

BagDownB3nny commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Sets checkMark as true
+     */
+    public void toBeMarked() {
+        this.checkMark = true;
+    }
+
+    /**
+     * Sets checkMark as false
+     */
+    public void toBeUnmarked() {
+        this.checkMark = false;
+    }
```
I think that renaming these functions as verbs might help make their purpose clearer

BagDownB3nny commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,72 @@
+     * or show items in a list and will exit if the bye command is returned
+     */
+    public void userInputs() {
+
```
Could this be renamed to something else to better reflect the purpose of this function? Maybe a verb like `initialise` could be better

BagDownB3nny reviewed (commented):
LGTM! Really enjoyed reviewing your code, as it was clean and well structured. Great work!

SHni99 committed:
Implement A-CheckStyle

joytqt-1202 commented on src/main/java/Duke.java:
```diff
@@ -1,6 +1,8 @@
 import java.util.*;
```
Maybe avoid using wild card imports

joytqt-1202 commented on src/main/java/Duke.java:
```diff
@@ -43,25 +46,32 @@ private static void userInputs() {
 
                 switch (instruction) {
+                    // Exit the system upon entering
                     case BYE:
```
Could you remove the indentation for switch cases to match the java coding conventions?
i.e.
```
switch(instructions) {
case BYE:
    ...
case LIST:
    ...
```

joytqt-1202 commented on src/main/java/Duke.java:
```diff
@@ -226,38 +229,66 @@ public String toString() {
+        this.endTime = endingPeriod[1];
     }
 
     @Override
```
For documentation purposes, you could make use of @inheritDoc for overriden public methods
i.e.
``` 
/**
 * {@inheritDoc}
 */
@Override
public...
```

joytqt-1202 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,41 @@
+import java.time.LocalDate;
+
+// Event class returns result that is type [E] and a starting time and an ending
+// time
```
Could make use of java docs notation to document code
i.e.
```
/**
 * Event class ...
 */
```

joytqt-1202 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toString() {
+        return (checkMark ? "[X] " : "[] ") + name;
```
I like your use of ternary conditional operators, over if-else!

SHni99 committed:
Implement branch-Level-10

joytqt-1202 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,176 @@
+
+
+/**
+ * Stores instructions and data in this class
```
I like your clear documentation for this class!

joytqt-1202 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,87 @@
+            index = Integer.parseInt(part[1]) - 1;
+        }
+
+        if (!Arrays.stream(Instructions.values()).anyMatch(x -> (x.toString()).equals(first_word))) {
```
Could you break down this conditional statement into smaller statements to avoid complicated code, so as to improve code readability?

joytqt-1202 reviewed (commented):
Enjoyed looking through your code! 
Gave me more insight into other forms of code development!

Just some comments on following Java Coding Conventions and some other code quality guidelines!

SHni99 committed:
Add javadoc to class

SHni99 committed:
Add exit feature

SHni99 committed:
Make changes to GUI display

SHni99 committed:
Create jar file using gradle

SHni99 committed:
1:Add assertion: assert to command in main duke class

Creates assert on command to prevent receiving null

To prevent it from getting null as input for command so that the method will end and return an error message before continuing to use the method in the command class. As assertion acts as handling exception also.

Let’s verify if command is null before executing its methods first

2:Add command: fix bug of entering the task without an item

User enters task types as input is accepted even though there is no item afterwards

Resolving this issue by adding warning output for users is better as it allows them to follow the input requirements set by the duke in order for the command to be write into the system correctly. Failure to do so it will return an alert message asking the users for input, until users enter task in correct format: [task type + task item] or exit the app.

Let’s change the return type of addItem method to string so that it will return a message upon entering an incorrect input

SHni99 committed:
Improve on coding quality: write cleaner and readable code

Allows other fellow reviewers to read the code at ease without having difficulties to understand. Allows me and others to find and fix any errors easier if the program does not work.

Accepting comments from reviewers in GitHub such as naming convention to improve methods and variables and adding inheritDoc in overridden methods to improve code readability and styling

Let’s check for styling and coding standards for each java file in duke, meanwhile also ensures the design of the program meets the code quality

SHni99 committed:
Merge pull request #1 from SHni99/branch-A-Assertions

Implement assert and streams in program

SHni99 committed:
Keep branch-A-CodeQuality up-to-date with master branch

SHni99 committed:
Add delay for app closing

SHni99 committed:
Merge pull request #2 from SHni99/branch-A-CodeQuality

Improve on styling and design of the Duke

SHni99 committed:
Add C-Update extension

SHni99 committed:
BCD-Extension: Add C-Update as a command to classes

BCD-Extension requires choosing at least one extension to add a function into the program

Implementation of updateTask method with parsing of index and input string to update the item in the list corresponding to its index. However, user are required to input the whole instruction in order for the program to return the correct outputs.

Implementing an update method helps to improve the user experience as users would find it easier to just update the item with one command, instead of deleting and re-adding of the item onto the list. This is also to retain the original list sequence.

SHni99 committed:
Improve on GUI

SHni99 committed:
Add title and colors for more personality

SHni99 committed:
Add screenshots of different commands

SHni99 committed:
Improve on code checkstyle

SHni99 committed:
Update README.md

SHni99 committed:
Add more JUnit testing

SHni99 committed:
Fix javafx to build gradle

SHni99 committed:
Add credit for code reuse before final release
